syntax = "proto3";
package trainerpb;

option csharp_namespace = "TrainerApi";

import "google/protobuf/timestamp.proto";

service TrainerService {
    rpc GetTrainer (TrainerByIdRequest) returns (TrainerResponse);
}

message TrainerByIdRequest {
    string id = 1;
}

message TrainerResponse {
    string id = 1;
    string name = 2;
    int32 age = 3;
    google.protobuf.Timestamp birthday = 4;
    repeated Medals medals = 5;
    google.protobuf.Timestamp created_at = 6;
}

message Medals {
    string region = 1;
    MedalType type = 2;
}

enum MedalType {
    UNKNOWN = 0;
    GOLD = 1;
    SILVER = 2;
    BRONZE = 3;
}